packages/scoped-elements/src/ScopedElementsMixin.js

```diff
@@ -86,20 +85,40 @@ const ScopedElementsMixinImplementation = superclass =>
       this.constructor.__registry = registry;
     }
 
-    /** @override */
     createRenderRoot() {
       const {
         scopedElements,
         shadowRootOptions,
         elementStyles,
       } = /** @type {typeof ScopedElementsHost} */ (this.constructor);
 
+      let supportsScopedRegistry = true;
       if (!this.registry) {
-        this.registry = new CustomElementRegistry();
-
-        Object.entries(scopedElements).forEach(([tagName, klass]) =>
-          this.registry.define(tagName, klass),
-        );
+        try {
+          this.registry = new CustomElementRegistry();
+        } catch (error) {
+          this.registry = customElements;
+          supportsScopedRegistry = false;
+        }
+        for (const [tagName, klass] of Object.entries(scopedElements)) {
+          const registeredClass = this.registry.get(tagName);
```

Comment:
I wonder if we should do the same check in the `defineScopedElement` method, as now they are sharing the same global registry and current implementation could cause silent issues.

Reply:
makes sense - will add 👍

Reply:
moved logic and added a test